core/google/cloud/iterator.py

```diff
@@ -69,6 +66,30 @@ class MyIterator(Iterator):
     2
     >>> iterator.page.remaining
     19
+
+It's also possible to consume an entire page and handle the paging process
+manually::
+
+    >>> iterator = MyIterator(...)
+    >>> items = list(iterator.page)
+    >>> items
+    [
+        <MyItemClass at 0x7fd64a098ad0>,
+        <MyItemClass at 0x7fd64a098ed0>,
+        <MyItemClass at 0x7fd64a098e90>,
+    ]
+    >>> iterator.page.remaining
+    0
+    >>> iterator.page.num_items
+    3
+    >>> iterator.next_page_token
+    'eav1OzQB0OM8rLdGXOEsyQWSG'
+    >>> # And just do the same thing to consume the next page.
+    >>> list(iterator.page)
+    [
+        <MyItemClass at 0x7fea740abdd0>,
+        <MyItemClass at 0x7fea740abe50>,
+    ]
```

Comment:
@jonparrott I'd like some advice here.

When `iterator.has_next_page()` is false, calling `iterator.page` will raise a `StopIteration`. This is the right move for `Iterator.next()` but not the right move for the snippet above.

Maybe the right answer is to have the `page` `@property` on `Iterator` stop throwing `StopIteration` and just have `Iterator.next` do it if `page` is `None`?